examples/gaussian_datatypes/example_explicit.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=import-outside-toplevel
+###############################################################################
+# Copyright (c), The AiiDA-CP2K authors.                                      #
+# SPDX-License-Identifier: MIT                                                #
+# AiiDA-CP2K is hosted on GitHub at https://github.com/aiidateam/aiida-cp2k   #
+# For further information on the license, see the LICENSE.txt file.           #
+###############################################################################
+"""Simple DFT calculations with Gaussian Datatypes examples"""
+
+import os
+import sys
+
+import click
+import ase.io
+
+from aiida.engine import run
+from aiida.orm import (Code, Dict, StructureData)
+from aiida.common import NotExistent
+
+from gdt_data import load_data
+
+
+def example_gdt(cp2k_code):
+    """
+    Testing CP2K with the Basis Set and Pseudopotential stored in gaussian.basisset/pseudo.
+    In this example the names of the basissets and pseudopotentials are still explicitly named
+    in the CP2K input section. The plugin will then validate the names against the data objects
+    passed to ensure consistency, and alter the generated CP2K input file to include the parameters
+    contained in the specified basisset and pseudopotential objects, rather than relying on the
+    data being present on the target node.
+    Use this mode if you are migrating from implicitly tracked basissets and pseudos or if you
+    you have more complicated setups (e.g. having different basissets for the same kind on different sites).
+    """
+
+    thisdir = os.path.dirname(os.path.realpath(__file__))
+
+    # Structure.
+    structure = StructureData(ase=ase.io.read(os.path.join(thisdir, '..', "files", 'h2o.xyz')))
+
+    # in your actual code you may want to get the basisset and pseudopotential
+    # with a lookup in the database, here we can simply call load_data()
+    bsets, pseudos = load_data(prefix="MY-EXPLICIT-")
```

Comment:
I tried to run the examples, but I got the following problem: when I run the first time, everything goes fine. However, if I run the second time, I get the following error:
```python
verdi run example_explicit.py cp2k
Traceback (most recent call last):
  File "/opt/conda/bin/verdi", line 8, in <module>
    sys.exit(verdi())
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/aiida/cmdline/utils/decorators.py", line 65, in wrapper
    return wrapped(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/aiida/cmdline/commands/cmd_run.py", line 147, in run
    exec(compile(handle.read(), scriptname, 'exec', dont_inherit=True), globals_dict)  # yapf: disable # pylint: disable=exec-used
  File "example_explicit.py", line 124, in <module>
    cli()  # pylint: disable=no-value-for-parameter
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "example_explicit.py", line 120, in cli
    example_gdt(code)
  File "example_explicit.py", line 81, in example_gdt
    "POTENTIAL": "{p.name}".format(p=pseudos["O"]),
KeyError: 'O'
```

for some reason the `bsets` and `pseudos` are empty.

Reply:
The cause for this is that you are re-running the example on a non-clean database and the BASIS_SET and PSEUDOPOTENTIAL have already been imported in the first run, combined with the duplicate-handling of `BasisSet.from_cp2k()`":

The default behavior of the `BasisSet.from_cp2k()` method  used to import basissets from a file is to ignore basissets with the same name already present in the database, meaning that if you do: `for b in BasisSet.from_cp2k(open("BASIS_MOLOPT")): b.store()` the first time it will store all basis sets from `BASIS_MOLOPT`, the second time `BasisSet.from_cp2k(open("BASIS_MOLOPT"))` will return an empty list instead since all of the basis sets are already there.

I fixed it now by doing a fallback on `BasisSet.get` (resp. `Pseudo.get`) when a `UniquenessError` gets thrown (with `duplicate_handling='error'`), but if that becomes a common use case (which I don't expect since most people will import the basis sets and pseudos once at the beginning of their journey with AiiDA), I can add another `duplicate_handling` strategy which transparently returns an already stored node instead.